src/librustc_trans/type_.rs

```diff
@@ -268,6 +269,20 @@ impl Type {
         }
     }
 
+    pub fn fat_ptr_data_type(&self) -> Type {
```

Comment:
Ok, just pulled that out into `type_of::fat_ptr_base_ty` since I need to call it from 2 different places and the second is not already within a match giving easy access to `mt.ty`.